src/filemap.rs

```diff
@@ -104,6 +105,17 @@ fn write_file(text: &StringBuffer,
             let stdout_lock = stdout.lock();
             try!(write_system_newlines(stdout_lock, text, config));
         }
+        WriteMode::Diff => {
+            println!("Diff of {}:\n", filename);
+            let mut f = try!(File::open(filename));
+            let mut ori_string = String::new();
+            try!(f.read_to_string(&mut ori_string));
+            let mut v = Vec::new();
+            try!(write_system_newlines(&mut v, text, config));
+            let text_str = String::from_utf8(v).unwrap();
+            let diff = make_diff(&ori_string, &text_str, 3);
```

Comment:
The order of original, formatted text is inverted compared to tests here. I don't know what the more natural choice is, but we should at least be consistent.